tests/no_version/test_unit_auth.py

```diff
@@ -55,3 +60,149 @@ async def test_verify_user_belongs_to_group(db):
     with pytest.raises(HTTPException) as exc_info:
         await _verify_user_belongs_to_group(user_id=1, user_group_id=42, db=db)
     debug(exc_info.value)
+
+
+async def test_check_project_dirs_update(local_resource_profile_db, db):
+    # Setup
+
+    resource, _ = local_resource_profile_db
+    # Add User
+    user = UserOAuth(
+        email="user@example.org",
+        hashed_password="12345",
+        project_dirs=[
+            "/example",
+            "/foo/bar",  # dataset1
+            "/test",
+            "/test/data",  # dataset2
+            "/test-1",  # dataset3
+        ],
+    )
+    db.add(user)
+    await db.commit()
+    await db.refresh(user)
+    # Add Project
+    project = ProjectV2(name="Project", resource_id=resource.id)
+    db.add(project)
+    await db.commit()
+    await db.refresh(project)
+    db.add(
+        LinkUserProjectV2(
+            project_id=project.id,
+            user_id=user.id,
+            is_owner=True,
+            is_verified=True,
+            permissions=ProjectPermissions.EXECUTE,
+        )
+    )
+    await db.commit()
+    # Add Datasets
+    dataset1 = DatasetV2(
+        name="Dataset 3",
+        project_id=project.id,
+        zarr_dir="/foo/bar/dataset/zarr",
+    )
+    dataset2 = DatasetV2(
+        name="Dataset 1",
+        project_id=project.id,
+        zarr_dir="/test/data/dataset/zarr",
+    )
+    dataset3 = DatasetV2(
+        name="Dataset 2",
+        project_id=project.id,
+        zarr_dir="/test-1/dataset/zarr",
+    )
+    db.add_all([dataset1, dataset2, dataset3])
+    await db.commit()
+    await db.refresh(dataset1)
+    await db.refresh(dataset2)
+    await db.refresh(dataset3)
+
+    kwargs = dict(
+        old_project_dirs=user.project_dirs,
+        user_id=user.id,
+        db=db,
+    )
+
+    # Test
```

Comment:
As a last check, I think we can play a bit more with trailing slashes.
The first example coming to mind is
```python
old_project_dirs = ["/data"]
new_project_dirs = []
some_zarr_dir = "/data/"
```

Expected outcome: 422